dbunit/src/test/java/com/baeldung/dbunit/DataSourceDBUnitTest.java

```diff
@@ -0,0 +1,121 @@
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import static com.baeldung.dbunit.ConnectionSettings.JDBC_URL;
+import static java.util.stream.Collectors.joining;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class DataSourceDBUnitTest extends DataSourceBasedDBTestCase {
+
+    @Override
+    protected DataSource getDataSource() {
+        JdbcDataSource dataSource = new JdbcDataSource();
+        dataSource.setURL(JDBC_URL);
+        dataSource.setUser("sa");
+        dataSource.setPassword("");
+        return dataSource;
+    }
+
+    @Override
+    protected IDataSet getDataSet() throws Exception {
+        return new FlatXmlDataSetBuilder().build(getClass()
+                .getClassLoader()
+                .getResourceAsStream("data.xml"));
+    }
+
+    @Override
+    protected DatabaseOperation getSetUpOperation() {
+        return DatabaseOperation.REFRESH;
+    }
+
+    @Override
+    protected DatabaseOperation getTearDownOperation() {
+        return DatabaseOperation.DELETE_ALL;
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        super.tearDown();
+    }
+
+
+    @Test
+    public void testSimpleDataSet() throws SQLException {
+        final Connection connection = getDataSource()
+                .getConnection();
+
+        final ResultSet rs = connection
+                .createStatement()
+                .executeQuery("select * from iTEMS where id = 1");
+
+        assertThat(rs.next()).isTrue();
+        assertThat(rs.getString("title")).isEqualTo("Grey T-Shirt");
+    }
+
+    @Test
+    public void testEmptySchema() throws Exception {
+        IDataSet expectedDataSet = getDataSet();
+        ITable expectedTable = expectedDataSet.getTable("CLIENTS");
+        IDataSet databaseDataSet = getConnection().createDataSet();
+        ITable actualTable = databaseDataSet.getTable("CLIENTS");
+        Assertion.assertEquals(expectedTable, actualTable);
+    }
+
+    @Test
+    public void testAssertByQuery() throws Exception {
+        IDataSet expectedDataSet = new FlatXmlDataSetBuilder().build(getClass()
+                .getClassLoader()
+                .getResourceAsStream("expected-user.xml"));
+        ITable expectedTable = expectedDataSet.getTable("CLIENTS");
+        Connection conn = getDataSource().getConnection();
+        conn.createStatement()
+                .executeUpdate(
+                        "INSERT INTO CLIENTS (first_name, last_name) VALUES ('John', 'Jansen')");
+        ITable actualData = getConnection()
+                .createQueryTable(
+                        "result_name",
+                        "SELECT * FROM CLIENTS WHERE last_name='Jansen'");
+        Assertion.assertEqualsIgnoreCols(expectedTable, actualData, new String[]{"id"});
+    }
+
+    @Test(expected = AssertionError.class)
```

Comment:
let's use at rule expected exception instead of the whole method check. this way we can granularly say when we expect the exception